bitnami/kube-prometheus/templates/prometheus/networkpolicy.yaml

```diff
@@ -0,0 +1,66 @@
+{{- /*
+Copyright VMware, Inc.
+SPDX-License-Identifier: APACHE-2.0
+*/}}
+
+# Based on upstream configuration: https://github.com/prometheus-operator/kube-prometheus/commit/ed15614b0a08482caa5cecb82f56356086d96bdf
+{{- if and .Values.prometheus.enabled .Values.prometheus.networkPolicy.enabled }}
+kind: NetworkPolicy
+apiVersion: {{ include "common.capabilities.networkPolicy.apiVersion" . }}
+metadata:
+  name: {{ template "kube-prometheus.prometheus.fullname" . }}
+  namespace: {{ include "common.names.namespace" . | quote }}
+  labels: {{- include "kube-prometheus.prometheus.labels" . | nindent 4 }}
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+spec:
+  podSelector:
+    matchLabels: {{- include "kube-prometheus.prometheus.matchLabels" . | nindent 6 }}
+  policyTypes:
+    - Ingress
+    - Egress
+  {{- if .Values.prometheus.networkPolicy.allowExternalEgress }}
+  egress:
+    - {}
+  {{- else }}
+  egress:
+    {{- if .Values.prometheus.networkPolicy.extraEgress }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.prometheus.networkPolicy.extraEgress "context" $ ) | nindent 4 }}
+    {{- end }}
+  {{- end }}
+  ingress:
+    {{/* TODO: Allow configuring prometheus container ports */}}
+    - ports:
+        - port: 9090
+        - port: 8080
+        {{- if .Values.prometheus.thanos.create }}
+        - port: 10901
+        - port: 10902
+        {{- end }}
```

Comment:
same here

Reply:
Prometheus ports are hardcoded in the operator. Added containerPorts for thanos